### DISCUSSION TITLE
Unloading of assemblies fails

### DISCUSSION THREAD

**Discussion by CAMongrel** [nxrighthere/UnrealCLR#190] [Help] [answered]
I'm tinkering around with developing a dedicated server which connects to a PostgreSQL database using Entity Framework Core. 
This is only a prototype, so the only other libraries my code is referencing are the EF core and EF core PostgreSQL ones.

Functionality-wise everything is fine, but I'm getting the following output when stopping playing in editor:
```
LogUnrealCLR: Warning: UnrealCLR::Module::Log: Unloading of assemblies took more time than expected. Trying to unload assemblies to the next breakpoint...
LogUnrealCLR: Error: UnrealCLR::Module::Log: Unloading of assemblies was failed! This might be caused by running threads, strong GC handles, or by other sources that prevent cooperative unloading.
```
The editor hangs for quite a few seconds before the messages appear.

I haven't dug too deep into the issue yet, but maybe you have an idea what may cause the EF core libraries to not unload properly? This is certainly related to the actual DB context, because the hang doesn't occur when I'm not doing any database interaction. I have a "using" around the DB context.

Any ideas or pointers what I should look into?

**Comment by nxrighthere** [comment, marked as answer]
This error indicates that one of the 3rd-party dependencies don't support unloadability. Please, see #110.

**Reply by nxrighthere** [reply]
In some cases, 3rd-party libraries provide functionality for manual disposal/deinitialization of resources in memory. You might have to invoke such functions from user code to let the garbage collector finish the rest.

**Reply by CAMongrel** [reply]
Thanks for your response and apologies for not searching hard enough for the other discussion. I'll check if there's something in the Postgres libraries to make this possible.
